### ISSUE TITLE
[BUG] Hls with codec avc1 on ios fail to play

### ISSUE THREAD

**Issue by bounty1342** [jhomlala/betterplayer#311]
**Describe the bug**
Error playing videos from pexel.com with only avc1 codec in m3u8 on iOS. 
Video is played fine on Safari or Chrome.

The error is the following :
flutter: ----------------FIREBASE CRASHLYTICS----------------
flutter: PlatformException(VideoError, Failed to load video: The network connection was lost., null, null)
flutter:
flutter: ----------------------------------------------------

**To Reproduce**

**Working file :**
https://www.pexels.com/video/two-boats-on-the-sea-besides-the-rocky-hills-island-above-sea-3120488/
```
#EXTM3U
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-medium",NAME="audio",AUTOSELECT=YES,DEFAULT=YES,CHANNELS="2",URI="https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/audio/0991f953/playlist.m3u8"
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-high",NAME="audio",AUTOSELECT=YES,DEFAULT=YES,CHANNELS="2",URI="https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/audio/0b40f588/playlist.m3u8"

#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=472067,AVERAGE-BANDWIDTH=458000,RESOLUTION=426x240,FRAME-RATE=24.000,CODECS="avc1.640815,mp4a.40.2",AUDIO="audio-medium"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/57465a68/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=706516,AVERAGE-BANDWIDTH=682000,RESOLUTION=640x360,FRAME-RATE=24.000,CODECS="avc1.64081E,mp4a.40.2",AUDIO="audio-medium"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/0991f953/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=1838025,AVERAGE-BANDWIDTH=1788000,RESOLUTION=960x540,FRAME-RATE=24.000,CODECS="avc1.64081F,mp4a.40.2",AUDIO="audio-medium"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/0b40f588/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=3346592,AVERAGE-BANDWIDTH=3097000,RESOLUTION=1280x720,FRAME-RATE=24.000,CODECS="avc1.640820,mp4a.40.2",AUDIO="audio-high"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/567e3457/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=6266979,AVERAGE-BANDWIDTH=5902000,RESOLUTION=1920x1080,FRAME-RATE=24.000,CODECS="avc1.640828,mp4a.40.2",AUDIO="audio-high"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/67938818/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=14259725,AVERAGE-BANDWIDTH=13727000,RESOLUTION=2560x1440,FRAME-RATE=24.000,CODECS="avc1.640832,mp4a.40.2",AUDIO="audio-high"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/f550519b/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=26008874,AVERAGE-BANDWIDTH=24930000,RESOLUTION=3840x2160,FRAME-RATE=24.000,CODECS="avc1.640834,mp4a.40.2",AUDIO="audio-high"
https://183vod-adaptive.akamaized.net/exp=1613767752~acl=%2Fe85ac9f7-fff2-4f29-80ab-8287bbdee986%2F%2A~hmac=04625103e2a99b89890d896f6f248a32f02ced1ff5a410510161e389464eff7f/e85ac9f7-fff2-4f29-80ab-8287bbdee986/sep/video/105189a8/playlist.m3u8

```

**Not working file :**
https://www.pexels.com/video/back-view-of-a-woman-854187/
```
#EXTM3U
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=339834,AVERAGE-BANDWIDTH=252000,RESOLUTION=426x226,FRAME-RATE=23.976,CODECS="avc1.640815"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/d1d32d38/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=558715,AVERAGE-BANDWIDTH=476000,RESOLUTION=640x338,FRAME-RATE=23.976,CODECS="avc1.64081E"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/7f8b173d/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=1691866,AVERAGE-BANDWIDTH=1395000,RESOLUTION=960x506,FRAME-RATE=23.976,CODECS="avc1.64081F"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/b84fc8fc/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=3003384,AVERAGE-BANDWIDTH=2752000,RESOLUTION=1366x720,FRAME-RATE=23.976,CODECS="avc1.640820"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/b6bfe376/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=6119191,AVERAGE-BANDWIDTH=5509000,RESOLUTION=2048x1080,FRAME-RATE=23.976,CODECS="avc1.640828"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/7426b8d7/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=14121524,AVERAGE-BANDWIDTH=13248000,RESOLUTION=2732x1440,FRAME-RATE=23.976,CODECS="avc1.640832"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/3c91ac96/playlist.m3u8
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=25483815,AVERAGE-BANDWIDTH=24349000,RESOLUTION=4096x2160,FRAME-RATE=23.976,CODECS="avc1.640834"
https://107vod-adaptive.akamaized.net/exp=1613768338~acl=%2F20ace0a1-d7f0-4219-b0c9-a959e09191a2%2F%2A~hmac=bd0543875cadd406f8fe238726f80c0f0ff56555355da03179025530e7ee8299/20ace0a1-d7f0-4219-b0c9-a959e09191a2/sep/video/ab5a5799/playlist.m3u8

```

**Flutter doctor**
```
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2.1 20D74 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Users/bounty1342/Documents/work/app/flutter
    • Framework revision 9b2d32b605 (4 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/bounty1342/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio
    • Android Studio at /Applications/Android Studio 4.2 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • sdk google atv x86 (mobile) • emulator-5554             • android-x86 • Android 10 (API 29) (emulator)
    • Not your iPhone (mobile)    • 00008020-001E7D503E86002E • ios         • iOS 14.4
```

**Better Player version**
- Version:0.0.55

**Smartphone (please complete the following information):**
 - Device: iPhoneXs
 - OS: iOS14.4

**Additional context**
Work great on Android.

Let me know if you need more informations.

PS: Also named constructor for BetterPlayerDataSource doesn't expose videoFormat option. (If is that doesn't change anything)
PS2: I can reproduce the issue from the comment https://github.com/jhomlala/betterplayer/issues/227#issuecomment-761035807, if the cache is set to true, it will then fail to play both video.

**Comment by jhomlala** [comment]
@bounty1342 Please provide direct links to these streams. I can't find them on these pages.

**Comment by bounty1342** [comment]
@jhomlala : You can try the one below

https://player.vimeo.com/external/363581079.m3u8?s=8b4ba71c76bf638e290977bc4a4a76d4c553205a&oauth2_token_id=57447763

https://player.vimeo.com/external/423177570.m3u8?s=993f95c4fa65c664a0a1ad2eae689370e5678f03&oauth2_token_id=56497713

PS : I changed the oauth2_token_id, but they still seems to download the m3u8.

**Comment by bounty1342** [comment]
The only difference I could find was :
CODECS="avc1.640828,mp4a.40.2"
CODECS="avc1.640828"

That's the reason I gave this issue this title, but it may be hard to spot on those m3u.

**Comment by jhomlala** [comment]
Fixed in 0.0.57.
